cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/generator/generic/FileInputConverter.java

```diff
@@ -51,17 +51,24 @@
                         "Could not deterime charset for file " + inputFile.getLocationURI() + " reading with UTF-8.");
                     charset = Charset.forName("UTF-8");
                 }
-
+                Path inputFilePath = Paths.get(inputFile.getLocationURI());
                 String readerType;
+
+                if (cobigen.isMostLikelyReadable("xml", inputFilePath)) {
+                    readerType = "xml";
+                    readAndAddInput(cobigen, convertedInputs, inputType, readerType, inputFilePath, charset);
+                    continue;
+                }
+
+                else if (cobigen.isMostLikelyReadable("openapi", inputFilePath)) {
+                    readerType = "openapi";
+                    readAndAddInput(cobigen, convertedInputs, inputType, readerType, inputFilePath, charset);
+                    continue;
+                }
                 {
```

Comment:
Can you check whether this bracket is needed? I think there are too many of them

Reply:
It is indeed not needed. However it war originally there, maybe for better code comprehension. 
Should I remove it ?

Reply:
In my opinion it should be removed, it made it harder for me to understand it. If you find it more readable like that, then you can leave it :)

Reply:
Okay then, they just got removed :)